.github/workflows/api-compatibility.yml

```
@@ -0,0 +1,127 @@
+                required: false
+                default: "2.8.58"
+
+permissions:
+    contents: read
+
+jobs:
+    api-compatibility:
+        runs-on: windows-latest
+        env:
+            # Default for PRs; workflow_dispatch can override via input
+            SER_VERSION: ${{ github.event_name == 'workflow_dispatch' && inputs.ser_version || '2.8.58' }}
+
+        steps:
+            - name: Checkout
+              uses: actions/checkout@v4
+              with:
+                  submodules: true
+
+            - name: Setup .NET
+              uses: actions/setup-dotnet@v4
+              with:
+                  dotnet-version: |
+                      8.x
+
+            # Install Rust toolchain on Windows for cargo prebuild
+            # - name: Install Rust (Windows)
+            #   shell: pwsh
+            #   run: |
+            #       $ProgressPreference = 'SilentlyContinue'
+            #       Invoke-WebRequest -Uri https://win.rustup.rs/x86_64 -OutFile rustup-init.exe
+            #       ./rustup-init.exe -y
+            #       echo "$env:USERPROFILE\.cargo\bin" | Out-File -FilePath $env:GITHUB_PATH -Encoding utf8 -Append
+            #       rustc -V
+            #       cargo -V
+
+            - name: Install dotnet-api-diff (Chocolatey)
+              shell: pwsh
+              run: |
+                  choco install dotnetapidiff --version=1.0.0 -y
+                  dotnetapidiff --version
+
+            # - name: Restore dependencies (for tooling)
+            #   shell: pwsh
+            #   run: |
+            #       dotnet restore benchmarks/csharp_benchmark.csproj
+
+            - name: Download StackExchange.Redis package (${{ env.SER_VERSION }})
+              shell: pwsh
+              run: |
+                  nuget install StackExchange.Redis -Version "$env:SER_VERSION" -OutputDirectory ./packages
+
+            - name: Build Valkey.Glide (net8.0)
+              shell: pwsh
+              run: |
+                  dotnet build --configuration Release --framework net8.0 --nologo
+
+            - name: Run API Diff (net8.0)
+              id: run-diff
+              shell: bash
+              run: |
+                  set -euo pipefail
+                  mkdir -p api-diff
+
+                  VALKEY_DLL="sources/Valkey.Glide/bin/Release/net8.0/Valkey.Glide.dll"
+                  SER_DLL="packages/StackExchange.Redis.${SER_VERSION}/lib/net8.0/StackExchange.Redis.dll"
+                  OUTFILE="api-diff/api-changes-net8.0.html"
+
+                  echo "Using StackExchange.Redis version: ${SER_VERSION}"
+                  echo "SER DLL: ${SER_DLL}"
+                  echo "Valkey DLL: ${VALKEY_DLL}"
+
+                  if [[ ! -f "$SER_DLL" ]]; then
+                    echo "# API Diff Error (net8.0)" > "$OUTFILE"
+                    echo "Could not find StackExchange.Redis DLL at $SER_DLL" >> "$OUTFILE"
+                    echo "exit_code=6" >> "$GITHUB_OUTPUT"
+                    exit 0
```

```suggestion
                    exit 1
```
pipeline failed - don't suppress it